src/deepsparse/v2/routers/router.py

```diff
@@ -13,61 +13,67 @@
 # limitations under the License.
 
 
-from typing import List, Tuple, Union
+from abc import abstractmethod
+from typing import Dict, List, Union
 
 from deepsparse.v2.operators import Operator
-from deepsparse.v2.schedulers import OperatorScheduler
-from deepsparse.v2.utils import Context, OperatorSchema
 
 
-__all__ = ["Router"]
+__all__ = ["Router", "LinearRouter"]
 
 
 class Router:
     """
-    Routers must implement a run method which runs a series of operators
-    for a pipeline for a given input. Base Router runs operators linearly
-    in a series
+    Routers dicate the next operator to run. Each Router must implement a next function,
+    which dictates the index or key of the next operator to run.
+
+    :param start_route: the start index or key of the router
+    :param end_route: the end index or key of the router
+
     """
 
-    @staticmethod
-    def run(
-        inp: OperatorSchema,
-        operators: List[Operator],
-        scheduler: OperatorScheduler,
-    ) -> Tuple[OperatorSchema, Context]:
+    def __init__(self, end_route: Union[str, int], start_route: Union[str, int]):
```

Comment:
One of. the arguments should also be `route: Union[Dict, List]` no?

Reply:
yup. updated in the follow-up PR